definitions/checks/check_for_newer_packages.rb

```diff
@@ -12,15 +12,54 @@ class Checks::CheckForNewerPackages < ForemanMaintain::Check
   end
 
   def run
-    exit_status, = ForemanMaintain.package_manager.check_update(packages: @packages,
-                                                                with_status: true)
-    assert(exit_status == 0, 'An update is available for one of these packages: '\
-                "#{@packages.join(',')}. Please update before proceeding.")
+    check_for_package_update
     if @manual_confirmation_version
       question = 'Confirm that you are running the latest minor release of Satellite '\
                 "#{@manual_confirmation_version}"
       answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
       abort! if answer != :yes
     end
   end
+
+  def check_for_package_update
+    exit_status, output = ForemanMaintain.package_manager.check_update(packages: @packages,
+                                                                       with_status: true)
+    packages_with_updates = []
+    if exit_status == 100
+      packages_with_updates = compare_pkg_versions(output).select do |_, result|
+        result == -1
+      end.keys
+    end
+
+    unless packages_with_updates.empty?
+      failure_message = 'An update is available for package(s): '\
+                        "#{packages_with_updates.join(',')}. Please update before proceeding!"
+      fail! failure_message
+    end
+  end
+
+  def packages_and_versions(output)
+    pkgs_versions = {}
+    pkg_details = output.split("\n\n")[1].split
+    iterator = 0
+    while iterator < 6
+      break if pkg_details[iterator].nil?
+      pkg_name = pkg_details[iterator].split('.').first
+      pkg_version = version(pkg_details[iterator + 1].split('-').first)
+      pkgs_versions[pkg_name] = pkg_version
+      iterator += 3
+    end
+    pkgs_versions
+  end
+
+  def compare_pkg_versions(output)
+    compare_pkg_versions = {}
+    packages_and_versions = packages_and_versions(output)
+    pkg_versions610 = { 'python3-pulp-2to3-migration' => version('0.12'),
+                        'tfm-rubygem-katello' => version('4.1') }
+    packages_and_versions.each do |name, version|
+      compare_pkg_versions[name] = version.<=>(pkg_versions610[name])
```

Comment:
Should this use `Gem::Version` and it's compare functionality? Comparing versions as strings does not always yield the expected results.

Reply:
The `packages_and_versions` forms the hash using `version` method which uses Gem::Version

Reply:
Ah, great, wasn't obvious to me. Thanks.